File: client/src/main/java/io/theam/client/commands/BaseCommandWithId.java
package io.theam.client.commands;

import com.github.rvesse.airline.annotations.Option;

public abstract class BaseCommandWithId extends BaseCommand {

    @Option(name = "--id", description = "Customer identity to handle")
    public Long customerId = -1L;

    @Override
    protected boolean validate() {
        boolean result = super.validate();
        if(customerId <  0L) {
            System.out.println("The customer id is necessary.");
            result = false;
        }
        return result;
    }
}

File: client/src/main/kotlin/es/nimio/exercise/client/commands/BaseCommandWithId.kt
package es.nimio.exercise.client.commands

import com.github.rvesse.airline.annotations.Option

abstract class BaseCommandWithId : BaseCommand() {

    @Option(name = ["--id"], description = "Customer identity to handle")
    var customerId: Long? = -1L

    override fun validate(): Boolean {
        var result = super.validate()
        if (customerId!! < 0L) {
            println("The customer id is necessary.")
            result = false
        }
        return result
    }
}
